### PR TITLE
Changes needed to build/test/release for 2.12.0

### PR DESCRIPTION
**Pull request by johanandren**

### CODE REVIEW

File: akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala [hunk 1]
```diff
@@ -39,11 +39,14 @@ class AddressUidExtension(val system: ExtendedActorSystem) extends Extension {
     else tlr.nextInt().toLong
   }
 
-  // used by old remoting and part of public api
-  @deprecated("Use longAddressUid instead", "2.4.x")
-  lazy val addressUid: Int = {
+  // private because GenJavaDoc fails on deprecated annotated lazy val
+  private lazy val _addressUid: Int = {
```
Comment (ktoso) [inline akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala]: Can we add a java `@Deprecated` maybe?
Hm the extension seems public though... will trigger MiMa, even if fine to do?

**Comment by ktoso** [comment]
LGTM

**Review by ktoso — APPROVED** [review]

**Comment by johanandren** [comment]
The what?! `java.lang.InternalError: Malformed class name`

**Comment by johanandren** [comment]
PLS BUILD

**Comment by johanandren** [comment]
Seems to be Scalacheck 1.13.4 not working with Scala 2.11, I'll make it conditional and choose 1.13.2 for 2.11 and 1.13.4 for 2.12 for now

**Comment by ktoso** [comment]
LGTM once validation is happy

File: project/Dependencies.scala [hunk 2]
```diff
@@ -17,19 +17,17 @@ object Dependencies {
   val sslConfigVersion = "0.2.1"
 
   val Versions = Seq(
-    crossScalaVersions := Seq("2.11.8"), // "2.12.0-RC2"
+    crossScalaVersions := Seq("2.11.8"), // "2.12.0"
```
Comment (patriknw) [inline project/Dependencies.scala]: @johanandren isn't it time that we add `2.12.0` to the `Seq` and actually cross publish all releases automatically from now? Is there still some dependencies that doesn't exist for 2.12? Can we define a different `crossScalaVersions` for those projects?

Comment (johanandren) [inline project/Dependencies.scala]: I'm not sure _why_ we had it like this, so I thought there were reasons(tm)

Comment (patriknw) [inline project/Dependencies.scala]: It was because we only released with best effort with those dependencies that were available for the milestone/rc.

**Pull request merged**